machine-learning/app/models/base.py

```diff
@@ -124,12 +125,15 @@ def _make_session(self, model_path: Path) -> AnnSession | ort.InferenceSession:
             case ".armnn":
                 session = AnnSession(model_path)
             case ".onnx":
+                cwd = os.getcwd()
+                os.chdir(model_path.parent)
                 session = ort.InferenceSession(
                     model_path.as_posix(),
                     sess_options=self.sess_options,
                     providers=self.providers,
                     provider_options=self.provider_options,
                 )
+                os.chdir(cwd)
```

Comment:
Wrap this in a try/finally so it always restores the original cwd even if there's an error.

Reply:
Makes sense